reports-backend/src/main/java/org/jboss/da/reports/api/AdvancedArtifactReport.java

```diff
@@ -0,0 +1,117 @@
+package org.jboss.da.reports.api;
+
+import org.jboss.da.common.version.VersionParser;
+import org.jboss.da.communication.model.GAV;
+import org.jboss.da.listings.api.service.WhiteArtifactService;
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Getter;
+
+public class AdvancedArtifactReport {
+
+    private WhiteArtifactService whiteArtifactService;
+
+    @Getter
+    private ArtifactReport artifactReport;
+
+    @Getter
+    private Set<GAV> blacklistArtifacts = new HashSet<>();
+
+    @Getter
+    private Set<GAV> whitelistArtifacts = new HashSet<>();
+
+    @Getter
+    private Set<GAV> communityGavsWithBestMatchVersions = new HashSet<>();
+
+    @Getter
+    private Set<GAV> communityGavsWithBuiltVersions = new HashSet<>();
+
+    @Getter
+    private Set<GAV> communityGavs = new HashSet<>();
+
+    public AdvancedArtifactReport(ArtifactReport artifactReport,
+            WhiteArtifactService whiteArtifactService) {
```

Comment:
If you need it just for the construction of AdvancedArtifactReport, I would pass as a methods parameter, not store it in field.

Or build the AdvancedArtifactReport from ouside, using some service.